subprojects/dependency-management/src/main/java/org/gradle/api/internal/artifacts/ivyservice/resolveengine/artifact/CompositeResolvedArtifactSet.java

```diff
@@ -30,6 +32,22 @@ private CompositeResolvedArtifactSet(List<ResolvedArtifactSet> sets) {
         this.sets = sets;
     }
 
+    public static List<ResolvedArtifactSet> unpack(ResolvedArtifactSet initialArtifactSet) {
```

Comment:
This is a `flatten()` operation, so we could call it that way. But maybe a visitor pattern would be more performant, as there's no need to copy stuff.